ebos/ecloutputblackoilmodule.hh

```diff
@@ -657,6 +665,36 @@ public:
                 cSalt_[globalDofIdx] = fs.saltConcentration().value();
             }
 
+            if (extboX_.size() > 0) {
+                extboX_[globalDofIdx] = intQuants.xvalue().value();
+            }
+
+            if (extboY_.size() > 0) {
+                extboY_[globalDofIdx] = intQuants.yvalue().value();
+            }
+
+            if (extboZ_.size() > 0) {
+                extboZ_[globalDofIdx] = intQuants.zFraction().value();
+            }
+
+            if (mFracCo2_.size() > 0) {
+                const Scalar stdVolOil = Opm::getValue(fs.saturation(oilPhaseIdx))*Opm::getValue(fs.invB(oilPhaseIdx))
+                                       + Opm::getValue(fs.saturation(gasPhaseIdx))*Opm::getValue(fs.invB(gasPhaseIdx))*Opm::getValue(fs.Rv());
+                const Scalar stdVolGas = Opm::getValue(fs.saturation(gasPhaseIdx))*Opm::getValue(fs.invB(gasPhaseIdx))*(1.0-intQuants.yvalue().value())
+                                       + Opm::getValue(fs.saturation(oilPhaseIdx))*Opm::getValue(fs.invB(oilPhaseIdx))*Opm::getValue(fs.Rs())*(1.0-intQuants.xvalue().value());
+                const Scalar stdVolCo2 = Opm::getValue(fs.saturation(gasPhaseIdx))*Opm::getValue(fs.invB(gasPhaseIdx))*intQuants.yvalue().value()
+                                       + Opm::getValue(fs.saturation(oilPhaseIdx))*Opm::getValue(fs.invB(oilPhaseIdx))*Opm::getValue(fs.Rs())*intQuants.xvalue().value();
+                const Scalar rhoO= 836.19;
+                const Scalar rhoG= 1.2984;
```

Comment:
This seems a bit hardcoded. The reference density should be available here.

Reply:
Ok